src/niamoto/core/components/exports/page_generator.py

```
@@ -686,3 +687,379 @@
+            # Logger l'erreur avec des informations détaillées
+            print(f"Failed to generate shape index page: {str(e)}")
+            import traceback
+
+            traceback.print_exc()
+
+            # Lever une erreur appropriée
+            raise TemplateError(
+                "shape_index.html",
+                f"Failed to generate shape index page: {str(e)}",
+                details={"error": str(e)},
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)